### PR TITLE
fix(proxy-rewrite): when conf.headers are missing,conf.method can make effect

### PR DESCRIPTION
**Pull request by liangliang4ward**
### What this PR does / why we need it:

when conf.headers are missing,conf.method can make effect

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tokers** [comment]
@liangliang4ward Better to add some test cases to cover it.

**Comment by liangliang4ward** [comment]
> The fix LGTM.
> 
> You can learn how to write tests via https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md and the current test
> 
> https://github.com/apache/apisix/blob/ebe91aa64c4fd26a03ea1ebddc089c577f4e730b/t/plugin/proxy-rewrite3.t#L87-L129

in proxy-rewrite3.t ,the `TEST 2` case according to the original code, it should not succeed。

**Comment by spacewander** [comment]
> > The fix LGTM.
> > You can learn how to write tests via [https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md](https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md?rgh-link-date=2022-02-14T01%3A06%3A06Z) and the current test
> > https://github.com/apache/apisix/blob/ebe91aa64c4fd26a03ea1ebddc089c577f4e730b/t/plugin/proxy-rewrite3.t#L87-L129
> 
> in proxy-rewrite3.t ,I thought the `TEST 2` case according to the original code, it should not succeed。

Let's add a test to prove your suspect. The test is passed in the CI.

**Comment by liangliang4ward** [comment]
> > > The fix LGTM.
> > > You can learn how to write tests via [https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md](https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md?rgh-link-date=2022-02-14T01%3A06%3A06Z) and the current test
> > > https://github.com/apache/apisix/blob/ebe91aa64c4fd26a03ea1ebddc089c577f4e730b/t/plugin/proxy-rewrite3.t#L87-L129
> > 
> > 
> > in proxy-rewrite3.t ,I thought the `TEST 2` case according to the original code, it should not succeed。
> 
> Let's add a test to prove your suspect. The test is passed in the CI.

ok，`TEST 2` case

**Comment by liangliang4ward** [comment]
> > > The fix LGTM.
> > > You can learn how to write tests via [https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md](https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md?rgh-link-date=2022-02-14T01%3A06%3A06Z) and the current test
> > > https://github.com/apache/apisix/blob/ebe91aa64c4fd26a03ea1ebddc089c577f4e730b/t/plugin/proxy-rewrite3.t#L87-L129
> > 
> > 
> > in proxy-rewrite3.t ,I thought the `TEST 2` case according to the original code, it should not succeed。
> 
> Let's add a test to prove your suspect. The test is passed in the CI.

this PR will to solve this issue; `TEST 2` case I don't why CI will success . this is my confusion.

**Comment by liangliang4ward** [comment]
> @liangliang4ward Better to add some test cases to cover it.

add a test case

File: t/plugin/proxy-rewrite3.t [hunk 1]
```diff
@@ -151,3 +151,57 @@ plugin_proxy_rewrite get method: GET
...
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 7: hit route(with out header)
```
Comment (spacewander) [inline t/plugin/proxy-rewrite3.t:198]:
```suggestion
=== TEST 7: hit route(with header)
```

File: t/plugin/proxy-rewrite3.t [hunk 2]
```diff
@@ -151,3 +151,57 @@ plugin_proxy_rewrite get method: GET
...
+--- request
+GET /hello HTTP/1.1
+--- response_body
+uri: /uri/plugin_proxy_rewrite
+host: apisix.iresty.com
+x-api-version: v1
+x-real-ip: 127.0.0.1
+--- grep_error_log_out
```
Comment (spacewander) [inline t/plugin/proxy-rewrite3.t:206]:
```suggestion
--- error_log
```
Now I know why the test can pass.

File: t/plugin/proxy-rewrite3.t (hunk 1 above)
Comment (liangliang4ward) [reply t/plugin/proxy-rewrite3.t]: modified

File: t/plugin/proxy-rewrite3.t (hunk 2 above)
Comment (liangliang4ward) [reply t/plugin/proxy-rewrite3.t]: ok, modified 'TEST 2' and 'TEST 4' case too

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

File: t/plugin/proxy-rewrite3.t [hunk 3]
```diff
@@ -151,3 +151,52 @@ plugin_proxy_rewrite get method: GET
 --- response_body
 property "method" validation failed: matches none of the enum values
 done
+
+
+
+=== TEST 6: set route(rewrite method with headers)
```
Comment (tokers) [inline t/plugin/proxy-rewrite3.t:157]: I think the test case should be "without headers"?

Comment (spacewander) [reply t/plugin/proxy-rewrite3.t:157]: There is a header in Line 173

Comment (liangliang4ward) [reply t/plugin/proxy-rewrite3.t:157]: > I think the test case should be "without headers"?

at line173 config rewrite header

**Comment by spacewander** [comment]
Merged. Thanks!

**Pull request merged**